src/player-server.ts

```diff
@@ -80,6 +80,22 @@ export default class PlayerServer implements IPlayable {
         }
     }
 
+    query(type: QueryType) {
+        return new Promise<any>((resolve, reject) => {
+            this.io.emit(
+                'query',
+                { type },
+                (err: string | null, value?: any) => {
+                    if (err) {
+                        return reject(err);
+                    } else {
+                        return resolve(value);
+                    }
+                },
+            );
+        });
+    }
+
```

Comment:
I mirrored your command system. Although I'm concerned with `CommandData`... I suggest to change both systems to discriminated unions, so that in the `switch (command.type)` cases, the data type would be automatically inferred. Safer and prettier!